setup.py

```diff
@@ -191,6 +192,12 @@ def cuda_ext_helper(name, sources, extra_cuda_flags, extra_cxx_flags=[]):
     extra_cxx_flags = ['-std=c++14', '-lcudart', '-lcublas', '-g', '-Wno-reorder', '-fopenmp', '-march=native']
     ext_modules.append(cuda_ext_helper('cpu_adam', ['cpu_adam.cpp'], extra_cuda_flags, extra_cxx_flags))
 
+if build_auto_ckpt_ext:
+    print(os.path.join(this_dir, "colossalai/fx/passes/algorithms/dynamic_programs.c"))
+    ext_modules.append(
+        Extension("c_version_dp",
```

Comment:
do not name this way, it is quite meaningless.

Reply:
I've fixed those problems, currently we will build the C extension the first time user run the rotor solver, only if C extension building fail will we use the python version of dynamic programming.